api/payload.go

```diff
@@ -26,40 +26,79 @@ import (
 // console. The proxy can output this data when asked for verbose output.
 //
 //  {
-//   "containerId": "756535dc6e9ab9b560f84c8...",
-//   "ctlSerial": "/tmp/sh.hyper.channel.0.sock",
-//   "ioSerial": "/tmp/sh.hyper.channel.1.sock"
+//    "containerId": "756535dc6e9ab9b560f84c8...",
+//    "ctlSerial": "/tmp/sh.hyper.channel.0.sock",
+//    "ioSerial": "/tmp/sh.hyper.channel.1.sock",
+//    "numIOStreams: 1
 //  }
 type RegisterVM struct {
 	ContainerID string `json:"containerId"`
 	CtlSerial   string `json:"ctlSerial"`
 	IoSerial    string `json:"ioSerial"`
 	Console     string `json:"console,omitempty"`
+	// NumIOStreams asks for a number of I/O tokens. An I/O token
+	// represents the communication between a container process inside
+	// the VM and a shim process outside the VM. This communication
+	// includes I/O streams (stdin, out, err) but also signals, exit
+	// status, ...
+	// The response frame will contain NumIOStreams I/O tokens.
+	NumIOStreams int `json:"numIOStreams,omitempty"`
+}
+
+// IOResponse is the response data in RegisterVMResponse and AttachVMResponse
+// when the client is asking for I/O tokens from the proxy (NumIOStreams > 0).
+type IOResponse struct {
+	// URL is the URL a shim process should connect to in order to initiate
+	// the I/O communication with the process inside the VM
+	URL string
```

Comment:
I understand but then it is not consistent with what we will get by using AttachVM. I mean that we will get a new URL for each container or process that we want to start on an existing pod, while we had the same URL for initial containers.
Maybe the solution is not to return any URL on AttachVM because we assume that RegisterVM has been performed before.